src/main/src/main/java/org/geoserver/security/impl/DataAccessRuleDAO.java

```diff
@@ -163,7 +164,9 @@ DataAccessRule parseDataAccessRule(String ruleKey, String ruleValue) {
         if(layerName != null) {
             if (!ANY.equals(root) && rawCatalog.getWorkspaceByName(root) == null)
                 LOGGER.warning("Namespace/Workspace " + root + " is unknown in rule " + rule);
-            if (!ANY.equals(layerName) && rawCatalog.getLayerByName(layerName) == null)
+            if (!ANY.equals(layerName) &&
+                    !(   rawCatalog.getLayerByName(new NameImpl(root, layerName)) != null
+                    || rawCatalog.getLayerGroupByName(root, layerName) != null ))
```

Comment:
Given the negation, wouldn't this be clearer? The double negation makes is a bit awkward to read at a glance.
```
if (!ANY.equals(layerName) &&
        rawCatalog.getLayerByName(new NameImpl(root, layerName)) == null
        && rawCatalog.getLayerGroupByName(root, layerName) == null )

```